### DISCUSSION TITLE
How to set default port number for app?

### DISCUSSION THREAD

**Discussion by dereckmezquita** [Appsilon/rhino#449] [Q&A] [answered]
I was using this:

```r
shiny::shinyApp(ui, server,
    options = list(
        port = 3003,
        launch.browser = TRUE
    )
)
```

How can I configure the default port number in `rhino`? I couldn't find the option in the `config.yml` nor `rhino.yml`. Should I create a new file `index.R` and then paste the above there and run that?

**Comment by kamilzyla** [comment, marked as answer]
Hi @dereckdemezquita! Thanks for using Rhino and for asking :slightly_smiling_face: Two approaches come to my mind:

### Set options in `.Rprofile`

Add these lines to the `.Rprofile` file in your project:
```r
options(
  shiny.port = 3003,
  shiny.launch.browser = TRUE
)
```

After restarting your R session, you can launch the app as usual by running `shiny::runApp()` in the R console or via "Run app" button (on the `app.R` file) in RStudio.

### Create a launch script

Create a `run.R` script with the following content:
```r
shiny::runApp(
  port = 3003,
  launch.browser = TRUE
)
```

You can then  launch the app by sourcing this script in your R console (`source("run.R")`). If you're on Linux / macOS, you can also make the script executable:
1. Put `#!/usr/bin/env R` in the first line of `run.R`.
2. Run `chmod +x run.R` in shell.
3. Launch it with `./run.R`.

**Reply by shahreyar-abeer** [reply]
Thanks @kamilzyla and @dereckdemezquita 
I ran into the same issue. This helps solve it!
